src/Gui/Tree.cpp

```diff
@@ -855,19 +860,35 @@ void TreeWidget::contextMenuEvent (QContextMenuEvent * e)
         DocumentObjectItem* objitem = static_cast<DocumentObjectItem*>
             (this->contextItem);
 
+        // check that the selection is not across several documents
+        bool acrossDocuments = false;
+        auto SelectedObjectsList = Selection().getCompleteSelection();
+        QList<QTreeWidgetItem*> items = this->selectedItems();
+        // get the object's document as reference
         App::Document* doc = objitem->object()->getObject()->getDocument();
+        for (auto it = SelectedObjectsList.begin(); it != SelectedObjectsList.end(); ++it) {
+            if ((*it).pDoc != doc)
+                acrossDocuments = true;
+        }
+                
         showHiddenAction->setChecked(doc->ShowHidden.getValue());
         contextMenu.addAction(this->showHiddenAction);
 
         hideInTreeAction->setChecked(!objitem->object()->showInTree());
         contextMenu.addAction(this->hideInTreeAction);
 
-        if (objitem->object()->getObject()->isDerivedFrom(App::DocumentObjectGroup::getClassTypeId()))
-            contextMenu.addAction(this->createGroupAction);
+        if (!acrossDocuments) { // is only sensible for selections within one document
+            if (objitem->object()->getObject()->isDerivedFrom(App::DocumentObjectGroup::getClassTypeId()))
+                contextMenu.addAction(this->createGroupAction);
+            contextMenu.addAction(this->selectAllChildrenAction);

```

Comment:
Can you check to see if it _has_ children first? The menu item doesn't make sense if there aren't any.

Reply:
OK, but this will change a lot. If there are childs, nothing will be added but I agree that we should have a context menu entry then. I'll think about it.

Reply:
Now this is the case, the context menu only appears if at least one item in the selection has children.